pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -613,8 +616,19 @@ public static void addColumnMetadataInfo(PropertiesConfiguration properties, Str
       properties.setProperty(getKeyFor(column, DATETIME_GRANULARITY), dateTimeFieldSpec.getGranularity());
     }
 
+    // complex field
+    if (fieldSpec.getFieldType().equals(FieldType.COMPLEX)) {
```

Comment:
(nit) Same for other enum checks
```suggestion
    if (fieldSpec.getFieldType() == FieldType.COMPLEX) {
```